lib/commands/backup.js

```diff
@@ -0,0 +1,200 @@
+            title: 'Checking configuration'
+        }, {
+            title: 'Backing up extensions',
+            task: this.backupExtensions
+        }, {
+            title: 'Backing up database',
+            task: this.backupDatabase
+        }, {
+            title: 'Backing up config files',
+            task: this.backupConfig
+        }, {
+            title: 'Backing up content folder',
+            task: this.backupContent
+        }, {
+            title: 'Writing backup file',
+            task: this.writeZipFile
+        }],{
+            env: this.system.environment,
+            argv: argv,
+            instance: instance,
+            hook: this.system.hook
+        });
+    }
+
+    initialize(ctx) {
+        const datetime = (new Date()).toJSON().substring(0, 10);
+
+        ctx.database = {};
+        ctx.saveLocation = process.cwd();
+
+        if (ctx.argv.output) {
+            ctx.saveLocation = absolutePath(ctx.argv.output);
+        }
+
+        try {
+            fs.ensureDirSync(ctx.saveLocation);
+            // After we're sure the directory exists, check to make sure we can write to it
+            fs.accessSync(ctx.saveLocation, fs.W_OK);
+        } catch (error) {
+            ctx.instance.ui.log('Ghost doesn\'t have permission to write to the output folder', 'red');
+            return Promise.reject(new errors.SystemError(error));
+        }
+
+        ctx.saveLocation = path.join(ctx.saveLocation, `${ctx.instance.name}.backup.${datetime}.zip`);
+
+        // The zip library used (adm-zip) has a known issue where the zip is structurally sound,
+        // but Windows has issues reading / unzipping it. To counteract this, all files are read
+        // and added via the `addFile` method, where the file permissions can be set, seemingly
+        // resolving the issue. See https://github.com/cthackers/adm-zip/issues/171
+        ctx.zipFile = new Zip();
+
+        if (ctx.instance.running()) {
+            ctx.instance.ui.log('Ghost is currently running. Backing up might take longer and slow down your site', 'yellow');
+            ctx.instance.loadRunningEnvironment();
+        } else {
+            ctx.instance.checkEnvironment();
+        }
+
+        return Promise.resolve();
+    }
+
+    backupContent(ctx) {
+        let files;
+        try {
+            files = walker('./content', {nodir: true});
+        } catch (error) {
+            debug(`Failed reading content folder: ${error}`);
+            return Promise.reject(new errors.SystemError({
+                context: error,
+                message: 'Failed to read content folder'
+            }));
+        }
+
+        const cwd = process.cwd();
+        const readFile = fs.readFileSync;
+
+        files.forEach((file) => {
+            const location = file.path.replace(/\\/g,'/');
+            let dir = path.dirname(location);
+            const name = path.basename(location);
+            dir = path.relative(cwd, dir);
```

Comment:
These few lines here will need to be updated to resolve symlinks - for instance the Casper theme is by default a symlink to `./current/content/themes/casper`.